migrated account fragment to jetpack compose

NiranjanNlc opened the pull request:
## Issue Fix
Fixes #1533 

## Screenshots
 

https://github.com/openMF/mobile-wallet/assets/25600880/06245532-ac08-4294-8cd5-8309e9cfbae9



## Description
 Migrated AccountFragments to jetpack compose screen 

##
<!--Please make sure these boxes are checked before submitting your pull request - thanks!-->

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio. 

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

PratyushSingh07 commented:
Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

PratyushSingh07 commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:
```diff
@@ -8,129 +8,42 @@ import android.view.View
-                DividerItemDecoration.VERTICAL
-            )
-        )
-        mRvLinkedBankAccounts!!.addOnItemTouchListener(
-            RecyclerItemClickListener(
-                activity,
-                object : RecyclerItemClickListener.OnItemClickListener {
-                    override fun onItemClick(childView: View?, position: Int) {
-                        val intent = Intent(
-                            activity,
-                            BankAccountDetailActivity::class.java
-                        )
-                        intent.putExtra(
-                            Constants.BANK_ACCOUNT_DETAILS,
-                            mBankAccountsAdapter!!.getBankDetails(position)
-                        )
-                        intent.putExtra(Constants.INDEX, position)
-                        startActivityForResult(intent, BANK_ACCOUNT_DETAILS_REQUEST_CODE)
-                    }
```
and have you handled this case in the new compose UI ?

NiranjanNlc replied:
oops i missed this , would implement it on next commit along with your comment on  desighn and binding .
Thanks ....

PratyushSingh07 commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:
```diff
@@ -8,129 +8,42 @@ import android.view.View
-    @JvmField
-    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
-    var tvTransactionsStateSubtitle: TextView? = null
+    lateinit var binding: FragmentAccountsBinding
```
can we get rid of binding ? with compose we shouldn't need this

NiranjanNlc replied:
it shall go while all the component of FinanceFragment is converted into jetpack compose.

NiranjanNlc commented:
> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design



> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design



> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

Its because of relative layout on the previous one and can you provide  me  some hint on how should i approach for the relative layout in jetpack compose .

NiranjanNlc commented:
> Hi @NiranjanNlc I think this UI is a bit different from the existing one . I would insist that you follow the original design

Changed the desighn and implemented the onClickListener as you suggested .

Please get it reviewed again and provide me suggestion.

NiranjanNlc commented:
@PratyushSingh07 ,
please go through the updated UI in PR description.
Hope it is upto the par.
Waiting for your review .

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt:
```diff
@@ -0,0 +1,146 @@
+import org.mifos.mobilewallet.mifospay.bank.viewmodel.BankAccountsViewModel
+
+@Composable
+fun AccountScreen(viewModel: BankAccountsViewModel,
```
Code not formatted, Please format the code.

NiranjanNlc replied:
done by ctrl+alt+L

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt:
```diff
@@ -0,0 +1,146 @@
+}
+
+@Composable
+fun AccountDisplay(
```
Rename it with AccountScreenContent

NiranjanNlc replied:
changed

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt:
```diff
@@ -0,0 +1,146 @@
+
+@Preview
+@Composable
+fun AccountScreenPreview()  {
```
Empty bank details preview is missing

NiranjanNlc replied:
created the new preview function for empty list

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AccountScreen.kt:
```diff
@@ -0,0 +1,146 @@
+        )
+    )
+    AccountDisplay(bankAccountDetails, {},{})
+//    AccountDisplay(bankAccountDetails = emptyList(),{})
```
remove this

NiranjanNlc replied:
done

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/AddAccountChip.kt:
```diff
@@ -0,0 +1,53 @@
+
+@Preview
+@Composable
+fun AddAccountChipPreview() {
```
wrap around the MifosTheme every preview function

NiranjanNlc replied:
done

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt:
```diff
@@ -0,0 +1,102 @@
+import org.mifos.mobilewallet.mifospay.R
+
+@Composable
+fun ItemCausualList(
```
ambiguous naming convention. what is Causual

therajanmaurya replied:
Write something related to feature

NiranjanNlc replied:
changed ListItemWithImage ..done

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/ItemCasualkList.kt:
```diff
@@ -0,0 +1,102 @@
+
+@Preview
+@Composable
+fun ComposeItemPreview() {
```
Preview name should be function name + Preview - suffix

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/composeScreen/PlaceHolderStateLayout.kt:
```diff
@@ -0,0 +1,56 @@
+import androidx.compose.ui.unit.dp
+import org.mifos.mobilewallet.mifospay.R
+@Composable
+fun PlaceholderStateLayout(
```
ambiguous naming

NiranjanNlc replied:
changed to EmptyStatePlaceHolder

therajanmaurya commented on mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/AccountsFragment.kt:
```diff
@@ -8,127 +8,48 @@ import android.view.View
+        binding= FragmentAccountsBinding.inflate(inflater, container, false)
+        ButterKnife.bind(this,binding.root)
+        binding.accountScreen.setContent {
+            AccountScreen(viewModel, ::addAccountClicked, ::onAccountClicked)
```
Don't need to pass the viewmodel init the ViewModel through hilt. 

See `CardsScreen` as an example.

NiranjanNlc replied:
done

NiranjanNlc committed:
migrated account fragment to compose

NiranjanNlc committed:
Merge branch 'dev' into accountFragmentToCompose

PratyushSingh07 commented:
closing this as completed by #1579
